newsroom/push.py

```diff
@@ -251,60 +257,69 @@ def get_event_dates(event):
-                                                                                            planning['guid']]}})
-        if orig_agendas.count() > 0:
-            orig_agenda = orig_agendas[0]
-
-        agenda = deepcopy(orig_agenda)
-        if not agenda:
-            # event id exists in planning item but event is not in the system
-            logger.warning('Event {} for planning {} couldn\'t be found'.format(planning['event_item'], planning))
-            # create new agenda
-            agenda = {}
-            init_adhoc_agenda(planning, agenda)
-        else:
-            if planning.get('state') in [WORKFLOW_STATE.CANCELLED, WORKFLOW_STATE.KILLED] or \
-                    planning.get('pubstatus') == 'cancelled':
-                # remove the planning item from the list
-                set_agenda_planning_items(agenda, orig_agenda, planning, action='remove')
+def publish_planning_item(planning, orig):
+    service = superdesk.get_resource_service("agenda")
+    agenda = deepcopy(orig)
+
+    init_adhoc_agenda(planning, agenda)
+
+    # Update agenda metadata
+    new_plan = set_agenda_metadata_from_planning(agenda, planning, force_adhoc=True)
 
-                service.patch(agenda['_id'], agenda)
-                return agenda
+    # Add the planning item to the list
+    set_agenda_planning_items(agenda, orig, planning, action="add" if new_plan else "update")
 
+    if not agenda.get("_id"):
+        # Setting ``_id`` of Agenda to be equal to the Planning item if there's no Event ID
+        agenda.setdefault("_id", planning["guid"])
+        agenda.setdefault("guid", planning["guid"])
+        return service.post([agenda])[0]
     else:
-        # there's no event item (ad-hoc planning item)
-        orig_agenda = app.data.find_one('agenda', req=None, _id=planning['guid']) or {}
-        agenda = deepcopy(orig_agenda)
-        init_adhoc_agenda(planning, agenda)
+        # Replace the original
+        service.patch(agenda["_id"], agenda)
+        return agenda["_id"]
+
+
+def publish_planning_into_event(planning) -> Optional[str]:
+    if not planning.get("event_item"):
+        return None
+
+    service = superdesk.get_resource_service('agenda')
 
-    # update agenda metadata
+    # this is a planning for an event item
+    # if there's an event then _id field will have the same value as event_id
+    orig_agendas = service.find(where={"_id": {"$in": [planning["event_item"], planning["guid"]]}})
+
+    if not orig_agendas.count():
+        # event id exists in planning item but event is not in the system
+        logger.warning('Event {} for planning {} couldn\'t be found'.format(planning['event_item'], planning))
+        return None
+
+    orig_agenda = orig_agendas[0]
+    agenda = deepcopy(orig_agendas[0])
+
+    if planning.get("state") in [WORKFLOW_STATE.CANCELLED, WORKFLOW_STATE.KILLED] or \
+            planning.get("pubstatus") == "cancelled":
```

Comment:
not sure what we use in planning, but in sd this is actually `canceled` as in [ninjs standard](http://www.iptc.org/std/ninjs/ninjs-schema_1.1.json)

Reply:
No worries, I'll double check the Event JSON formatter

Reply:
I don't know the reason why, but in Planning we use `cancelled`:
* https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/common.py#L71
* https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/common.py#L78
* https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/planning/planning.py#L1552

Reply:
I see, we should harmonize that eventually but that's for another PR